File: 5-hash/hash/src/main/java/com/mycompany/hash/Aluno.java

package com.mycompany.hash;

/**
 *
 * @author alexandrezamberlan
 */
public class Aluno {
    public int matricula;
    public String nome;

    public Aluno(int matricula, String nome) {
        this.matricula = matricula;
        this.nome = nome;
    }
    
    public Aluno(int matricula) {
        this.matricula = matricula;
    }

    @Override
    public int hashCode() {
        int hash = 7;
        hash = 41 * hash + this.matricula;
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Aluno other = (Aluno) obj;
        if (this.matricula != other.matricula) {
            return false;
        }
        return true;
    }

    @Override
    public String toString() {
        return "Aluno{" + "matricula=" + matricula + ", nome=" + nome + '}';
    }
    
    
    
    
}

File: 04-pesquisa/PesquisaEmColecoesOO/PesquisaTeste/Aluno.cs
public class Aluno {
    public int Matricula {get;set;}
    public string Nome {get;set;}

    public Aluno(int m, string n) {
        Matricula = m;
        Nome = n;
    }

    public override string ToString()
    {
        return "Nome: " + Nome + ". Matrícula: " + Matricula;
    }

    public override bool Equals(object? o)
    {
        
        if (o.GetType() == GetType()) 
        {
            Aluno e = (Aluno) o;
            return this.Matricula == e.Matricula;
        }
        return false; 
    }
}